### PR TITLE
test(create-gatsby): add test for handling errors in install plugins

### PR DESCRIPTION
**Pull request by mfrachet**
## Description

Unit test to catch and lock error on the the install plugins step of the create-gatsby cli

### CODE REVIEW

File: packages/create-gatsby/src/require-utils.ts [hunk 1]
```diff
@@ -0,0 +1,4 @@
+export const requireResolve = (
```
Comment (mfrachet) [inline packages/create-gatsby/src/require-utils.ts:1]: ℹ️ Aims to mock `require.resolve` that is not yet chipped in jest: https://github.com/facebook/jest/issues/9543

File: packages/create-gatsby/src/__tests__/install-plugins.ts [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+      return `somewhere-i-belong`
+    })
+
+    await installPlugins([], {}, `not-existing-path`, [])
+
+    // Test function behaviour but improves the DX, it probably worth it
+    expect(reporter.error).toBeCalledWith(
+      `gatsby-cli not installed, or is too old`
+    )
+  })
```
Comment (gillkyle) [inline packages/create-gatsby/src/__tests__/install-plugins.ts:63]: It looks like the only thing that's different about this test from the one above it is the test name  and the error message 🤔  am I missing something?

**Review by gillkyle — COMMENTED** [review]
I think this looks good but I see one test that looks like maybe it's duplicated? Not sure what the exact case is there.

File: packages/create-gatsby/src/__tests__/install-plugins.ts (hunk 2 above)
Comment (mfrachet) [reply packages/create-gatsby/src/__tests__/install-plugins.ts:63]: That's a great catch! The setup of this scenario is not made correctly, I need to fix this 😊

**Review by ascorbic — APPROVED** [review]
Looking good!

**Pull request merged**